src/Formats/FormatFactory.h

```diff
@@ -266,7 +266,7 @@ class FormatFactory final : private boost::noncopyable
 
 private:
     FormatsDictionary dict;
-    FileExtensionFormats file_extension_formats;
+    FileExtensionFormats file_extension_formats;    // Also used as a case-insensitive format_name mapping.
```

Comment:
I think it will be better to just use lower case names in `FormatsDictionary dict`, and always search by lower names. We can use `to_lower` in `getCreators` and use `getCreators` instead of all `dict.at(name)` usages. Also we can add new method `FormatFactory::Creators & FormatFactory::getOrCreateCreators(const String & name)` that will also use lover name, but will be used instead of `dict[name]`

Reply:
1. Using lowercase directly causes the name column in the `system.forms` table to be all lowercase. In `StorageSystemFormats::fillData()`, directly use the key of the dict as the format column
2. Many of the functional tests use format column for queries (such as, `select number_of_rows from system.schema_inference_cache where format='Npy'` in 02908_Npy_files_caching), so it is necessary to use original format name

Reply:
Do you mean using another map (maybe `std::unordered_map<String, String> loverToLower`) to store the lover name for display in `system.formats`?

Reply:
> Using lowercase directly causes the name column in the system.formats table to be all lowercase. In StorageSystemFormats::fillData(), directly use the key of the dict as the format column

True, didn't think about it, thanks for pointing. So, we should keep original format names for such cases. Using `file_extension_formats` for matching is not good, this map is just not supposed to be used for it (and by it's name it's not clear why it contains all format names in lower case).
I think the best option is to use lower case names in `dict` and store original format name in the `Creators` struct as a separate field and use it when we need original name. As an alternative, we can add a separate map for matching original and lower case format names, but first option sounds better for me